concordium-consensus/src/Concordium/GlobalState.hs

```diff
@@ -416,13 +432,24 @@ class GlobalStateConfig (c :: Type) where
 
 instance GlobalStateConfig MemoryTreeMemoryBlockConfig where
     type GSContext MemoryTreeMemoryBlockConfig pv = ()
-    type GSState MemoryTreeMemoryBlockConfig pv = SkovData pv (BS.HashedBlockState pv)
+    type GSState MemoryTreeMemoryBlockConfig pv = Basic.SkovData pv (BS.HashedBlockState pv)
     initialiseExistingGlobalState _ _ = return Nothing
+
+    migrateExistingState MTMBConfig{..} () Basic.SkovData{..} migration regen = do
+        case _nextGenesisInitialState of
+            Nothing -> error "Precondition violation. The initial state must exist."
+            Just bs ->
+                case migrateBlockState migration (_unhashedBlockState bs) of
+                    Left err -> error $ "Precondition violation. Cannot migrate existing state: " ++ err
+                    Right newbs -> do
+                        skovData <- runPureBlockStateMonad (Basic.initialSkovData mtmbRuntimeParameters (regenesisConfiguration regen) (BS.hashBlockState newbs) _transactionTable (Just _pendingTransactions))
```

Comment:
Sigh. This is true, but annoying and irrelevant since this state is not used for anything meaningful and is going to be removed very soon.

Anyhow, I added removal of finalized transactions from the transaction table.